sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/FileConfiguration.java

```diff
@@ -49,7 +49,7 @@ public final class FileConfiguration {
 
   private static final Logger logger = Logger.getLogger(FileConfiguration.class.getName());
   private static final Pattern ENV_VARIABLE_REFERENCE =
-      Pattern.compile("\\$\\{([a-zA-Z_][a-zA-Z0-9_]*)}");
+      Pattern.compile("\\$\\{(?<ENVKEY>[a-zA-Z_][a-zA-Z0-9_]*)(:-(?<DEFAULTVALUE>[^\n]*))?\\}");
```

Comment:
These regex group names aren't referenced below...do you like them purely as embedded documentation or want to remove them or?

Reply:
Ah that's right. I had been using them but animalsniffer told me android doesn't support accessing groups by name so I had to revert to index. Kind of disappointing because referencing by name is a lot more clear.